### PR TITLE
Serialize: set empty string if value function returns null

### PR DESCRIPTION
**Pull request by jtrumbull**
https://github.com/jquery/jquery/issues/3005

### CODE REVIEW

**Comment by mgol** [comment]
@jtrumbull Thanks for the PR! This will need unit tests to prove that this PR fixes the linked issue. Tests for this module are located in https://github.com/jquery/jquery/blob/master/test/unit/serialize.js.

**Comment by jtrumbull** [comment]
@mgol Full test suite is passing.

File: test/unit/serialize.js [hunk 1]
```diff
@@ -73,13 +73,16 @@ QUnit.test( "jQuery.param()", function( assert ) {
...
+	params = { "param1": function() {}, "param2": function() { return null; } };
+	assert.equal( jQuery.param( params, false ), "param1=&param2=", "object with method properties return null" );
+	
 	params = { "test": { "length": 3, "foo": "bar" } };
 	assert.equal( jQuery.param( params, false ), "test%5Blength%5D=3&test%5Bfoo%5D=bar", "Sub-object with a length property" );
 
 	params = { "test": [ 1, 2, null ] };
 	assert.equal( jQuery.param( params, false ), "test%5B%5D=1&test%5B%5D=2&test%5B%5D=", "object with array property with null value" );
-
+
```
Comment (mgol) [inline test/unit/serialize.js]: Please revert the change in this line. We disallow trailing spaces.

Comment (mgol) [inline test/unit/serialize.js]: This applies to a few other changed lines as well.

**Comment by jtrumbull** [comment]
@mgol I fixed the whitepace issue. However, this made me curious as to why an error wasn't being thrown by `jscs`. After looking at the gruntfile, I noticed that it's because only parts of the test directory are being run through. More out of curiosity than anything, do you happen to know why this is?

**Comment by mgol** [comment]
Ideally we'd check every file but those test files are very old, they contain lots of errors so fixing them is not an easy task... We're enabling JSCS file by file but we've got a long way ahead to lint everything.

Source is linted completely, though, only tests are lacking.

File: src/serialize.js [hunk 2]
```diff
@@ -58,7 +58,8 @@ jQuery.param = function( a, traditional ) {
 		add = function( key, value ) {
 
 			// If value is a function, invoke it and return its value
-			value = jQuery.isFunction( value ) ? value() : ( value == null ? "" : value );
+			value = jQuery.isFunction( value ) ? value() : value;
+			value = value == null ? "" : value;
```
Comment (gibson042) [inline src/serialize.js]: This may be smaller as `if ( value == null )`.

File: test/unit/serialize.js [hunk 3]
```diff
@@ -74,6 +74,9 @@ QUnit.test( "jQuery.param()", function( assert ) {
 	params = { "param1": null };
 	assert.equal( jQuery.param( params, false ), "param1=", "Make sure that null params aren't traversed." );
 
+	params = { "param1": function() {}, "param2": function() { return null; } };
+	assert.equal( jQuery.param( params, false ), "param1=&param2=", "object with method properties return null" );
```
Comment (mgol) [inline test/unit/serialize.js]: `object with method properties return null` -> `Object with method properties that return null`

**Comment by mgol** [comment]
LGTM.
